sdk/cosmosdb/Azure.ResourceManager.CosmosDB/tests/ScenarioTests/DatabaseAccountOperationsTests.cs

```diff
@@ -0,0 +1,257 @@
+
+namespace Azure.ResourceManager.CosmosDB.Tests
+{
+    [TestFixture]
+    public class DatabaseAccountOperationsTests : CosmosDBManagementClientBase
+    {
+        private string resourceGroupName;
+        private const string databaseAccountName = "db4937";
+        private const int maxStalenessPrefix = 300;
+        private const int maxIntervalInSeconds = 1000;
+        public DatabaseAccountOperationsTests() : base(true)
+        {
+        }
+
+        [SetUp]
+        public async Task ClearAndInitialize()
+        {
+            if (Mode == RecordedTestMode.Record || Mode == RecordedTestMode.Playback)
+            {
+                InitializeClients();
+                this.resourceGroupName = Recording.GenerateAssetName(CosmosDBTestUtilities.ResourceGroupPrefix);
+                await CosmosDBTestUtilities.TryRegisterResourceGroupAsync(ResourceGroupsOperations,
+                    CosmosDBTestUtilities.Location,
+                    this.resourceGroupName);
+            }
+        }
+
+        [TearDown]
+        public async Task CleanupResourceGroup()
+        {
+            await CleanupResourceGroupsAsync();
+        }
+
+        [TestCase]
+        public async Task DatabaseAccountCRUDTest()
+        {
+            var locations = new List<Location>();
+            locations.Add(new Location(id: null, locationName: "WEST US", documentEndpoint: null, provisioningState: null, failoverPriority: null, isZoneRedundant: false));
+            var databaseAccountCreateOrUpdateParameters = new DatabaseAccountCreateUpdateParameters(locations);
+            databaseAccountCreateOrUpdateParameters.Location = "WEST US";
+            databaseAccountCreateOrUpdateParameters.Tags.Add("key1", "value1");
+            databaseAccountCreateOrUpdateParameters.Tags.Add("key2", "value2");
+            databaseAccountCreateOrUpdateParameters.Kind = DatabaseAccountKind.MongoDB;
+            databaseAccountCreateOrUpdateParameters.ConsistencyPolicy = new ConsistencyPolicy(DefaultConsistencyLevel.BoundedStaleness, maxStalenessPrefix, maxIntervalInSeconds);
+            databaseAccountCreateOrUpdateParameters.IpRules.Add(new IpAddressOrRange("23.43.230.120"));
+            databaseAccountCreateOrUpdateParameters.IsVirtualNetworkFilterEnabled = true;
+            databaseAccountCreateOrUpdateParameters.EnableAutomaticFailover = false;
+            databaseAccountCreateOrUpdateParameters.ConnectorOffer = "Small";
+            databaseAccountCreateOrUpdateParameters.DisableKeyBasedMetadataWriteAccess = false;
+            DatabaseAccountGetResults databaseAccountGetResults1 =
+                await WaitForCompletionAsync(
+                    await CosmosDBManagementClient.DatabaseAccounts.StartCreateOrUpdateAsync(resourceGroupName, databaseAccountName, databaseAccountCreateOrUpdateParameters));
+            Assert.AreEqual(200, CosmosDBManagementClient.DatabaseAccounts.CheckNameExistsAsync(databaseAccountName).Result.Status);
```

Comment:
Do not do *.Result since you risk deadlock in certain situations

https://www.keuperict.nl/posts/development/2018/11/28/deadlocks-with-async-and-await/